### PR TITLE
Change group info url structure to match web api

### PR DESCRIPTION
**Pull request by CalumMcCall**
- PRO-1365

### CODE REVIEW

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupInfo.scala [hunk 1]
```diff
@@ -31,10 +33,23 @@ object GroupInfo extends SlickJsonHelper {
 
   val groupInfos = TableQuery[GroupInfoTable]
 
-  protected def getGroup(groupName: String, namespace: Namespace) =
+  case class ValidName()
+
+  type Name = Refined[String, ValidName]
+
+  implicit val validGroupName: Validate.Plain[String, ValidName] =
```
Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupInfo.scala]: why was this added?

File: device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala [hunk 2]
```diff
@@ -122,48 +120,45 @@ class Routes(namespaceExtractor: Directive1[Namespace],
   def api: Route =
```
Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: btw all these routes should have their own resource, ie, system info resource and group info resource. 

Also this file should not be named "Routes", that means nothing.

But I guess this can be done in another ticket

File: device-registry/src/main/scala/org/genivi/sota/device_registry/GroupInfo.scala (hunk 1 above)
Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/GroupInfo.scala]: It seemed sensible to put some constraints on what a groupName can be. I wanted to extract groupName from inside the url path, to make it consistent with the system_info routes.

File: device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala (hunk 2 above)
Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: Nah, I can have a go at doing that refactoring here, might as well.

Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: After having a look, I think it probably makes sense to split system_info and group_info into separate prefixes, after "devices". It would also then make sense to change the url structure on web. So I think making a separate ticket for it makes sense, as it will be a fairly big change.

Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: yeah we should do that as well...

Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: wait what do you mean after devices? `system_info` is a resource of a device resource, so it makes sense that it's after a device id

Comment (CalumMcCall) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: I've created a ticket for it, PRO-1411.

Comment (simao) [inline device-registry/src/main/scala/org/genivi/sota/device_registry/Routes.scala]: Ok but not sure about that.

**Comment by simao** [comment]
11/11
:+1:
